cookbook/controller/argument_value_resolver.rst

```diff
@@ -0,0 +1,230 @@
+    has a priority of 100. As this  one is responsible for fetching attributes
+    from the ``Request``, it's also recommended to trigger your custom value
+    resolver with a lower priority. This makes sure the argument resolvers
+    are not triggered in (e.g.) subrequests if you pass your user along:
+    ``{{ render(controller('AppBundle:User:index', {'user', app.user})) }}``.
+
+.. configuration-block::
+
+    .. code-block:: yaml
+
+        # app/config/services.yml
+        services:
+            app.value_resolver.user:
+                class: AppBundle\ArgumentValueResolver\UserValueResolver
+                arguments:
+                    - '@security.token_storage'
+                tags:
+                    - { name: controller.argument_value_resolver, priority: 50 }
+
+    .. code-block:: xml
+
+        <!-- app/config/services.xml -->
+        <?xml version="1.0" encoding="UTF-8" ?>
+        <container xmlns="http://symfony.com/schema/dic/services"
+            xmlns:xsi="'http://www.w3.org/2001/XMLSchema-Instance"
+            xsi:schemaLocation="http://symfony.com/schema/dic/services http://symfony.com/schema/dic/services/services-1.0.xsd">
+
+            <services>
+                <service id="app.value_resolver.user" class="AppBundle\ArgumentValueResolver\UserValueResolver">
+                    <argument type="service" id="security.token_storage">
+                    <tag name="controller.argument_value_resolver" priority="50" />
+                </service>
+            </services>
+
+        </container>
+
+    .. code-block:: php
+
+        // app/config/services.php
+        use Symfony\Component\DependencyInjection\Definition;
+
+        $defintion = new Definition(
+            'AppBundle\ArgumentValueResolver\UserValueResolver',
+            array(new Reference('security.token_storage'))
+        );
+        $definition->addTag('controller.argument_value_resolver', array('priority' => 50));
+        $container->setDefinition('app.value_resolver.user', $definition);
+
+.. _`yield`: http://php.net/manual/en/language.generators.syntax.php
+
+Creating an Optional User Resolver
+----------------------------------
+
+When you want your user to be optional, e.g. when your page is behind a
+firewall that also allows anonymous authentication, you might not always
+have a security user. To get this to work, you only have to change your
+method signature to `UserInterface $user = null`.
+
+When you take the ``UserValueResolver`` from the previous example, you can
+see there is no logic in case of failure to comply to the requirements. Default
+values in are defined in the signature and are available in the ``ArgumentMetadata``.
+When a default value is available and there are no resolvers that support
+the given value, the ``DefaultValueResolver`` is triggered. This Resolver
+takes the default value of your argument and yields it to the argument list::
```

Comment:
There is no "yield" that I can use though, the only word here referring to it is "yields"